tests/unit/imports/server/publishJoinedQuery.ts

```diff
@@ -0,0 +1,174 @@
+import { Meteor, Subscription } from 'meteor/meteor';
+import { Mongo, MongoInternals } from 'meteor/mongo';
+import { Random } from 'meteor/random';
+import { assert } from 'chai';
+import Guesses from '../../../../imports/lib/models/Guesses';
+import Hunts from '../../../../imports/lib/models/Hunts';
+import Puzzles from '../../../../imports/lib/models/Puzzles';
+import Tags from '../../../../imports/lib/models/Tags';
+import makeFixtureHunt from '../../../../imports/server/makeFixtureHunt';
+import publishJoinedQuery from '../../../../imports/server/publishJoinedQuery';
+
+const { MongoError } = MongoInternals.NpmModules.mongodb.module;
+
+class StubSubscription implements Subscription {
+  stopHooks: (() => void)[] = [];
+
+  onStop(func: () => void): void {
+    this.stopHooks.push(func);
+  }
+
+  data: Map<
+    string /* collection */,
+    Map<string /* id */, Record<string, any> /* fields */>
+  > = new Map();
+
+  added(collection: string, id: string, fields: Record<string, any>): void {
+    let collectionMap = this.data.get(collection);
+    if (!collectionMap) {
+      collectionMap = new Map();
+      this.data.set(collection, collectionMap);
+    }
+    collectionMap.set(id, fields);
+  }
+
+  changed(collection: string, id: string, fields: Record<string, any>): void {
+    const collectionMap = this.data.get(collection);
+    if (!collectionMap) {
+      throw new Error(`Collection ${collection} not found`);
+    }
+    const doc = collectionMap.get(id);
+    if (!doc) {
+      throw new Error(`Document ${id} not found in collection ${collection}`);
+    }
+    collectionMap.set(id, { ...doc, ...fields });
+  }
+
+  removed(collection: string, id: string): void {
+    const collectionMap = this.data.get(collection);
+    if (!collectionMap) {
+      throw new Error(`Collection ${collection} not found`);
+    }
+    if (!collectionMap.delete(id)) {
+      throw new Error(`Document ${id} not found in collection ${collection}`);
+    }
+  }
+
+  ready() { /* noop */ }
+
+  stop() { /* noop */ }
```

Comment:
Arguably this should call each of the `stopHooks`?  Then maybe the `after()` calls would just call `stop()` here.

Reply:
Yeah, reasonable. Fixed.